### PR TITLE
import test suite from ranmocy

### PR DESCRIPTION
**Pull request by paulirish**
updated. now works, thanks to puppeteer. 

run `yarn test` and it'll scrape some json feeds from the website and then test them against the page results and see that it all is correct.

image

### CODE REVIEW

File: javascripts/default.js [hunk 1]
```diff
@@ -476,4 +481,196 @@ NodeList.prototype.on = NodeList.prototype.addEventListener = (function(name, fn
...
+    Object.keys(name_to_path).forEach(function(name) {
+      fetch(name_to_path[name]).then(function(r){ return r.json(); }).then(function(json) {
+        data[name] = json;
+        for (var p in data) {
+          if (typeof(data[p]) === 'undefined') {
+            // not all finished, ignore
+            return;
+          }
+        }
+        callback(data);
```
Comment (samccone) [inline javascripts/default.js]: callback will be invoked N times where N is the number of keys in name_to_path... is this by design?

File: Gemfile [hunk 2]
```diff
@@ -3,3 +3,8 @@ source 'https://rubygems.org'
 gem 'rake'
 gem 'pry'
 gem 'plist'
+
+gem 'nokogiri'
```
Comment (samccone) [inline Gemfile]: wait, we are pulling into ruby testing tools but I see no new tests

File: javascripts/default.js (hunk 1 above)
Comment (paulirish) [reply javascripts/default.js]: good catch. fixed.

**Comment by paulirish** [comment]
updated. `yarn test` to do the whole thing.

**Comment by samccone** [comment]
boom nice

**Pull request merged**